src/utils/mmbot_strong_types.hpp

```diff
@@ -36,28 +36,68 @@ namespace antara
             st::addable_with<char *>,
             st::addable_with<const char *>>;
 
-    using st_quote = st::type<
+    using st_symbol = st::type<
             std::string,
-            struct quote_tag,
+            struct symbol_tag,
             st::equality_comparable,
             st::addable_with<char *>,
             st::addable_with<const char *>>;
 
-    using st_base = st::type<
-            std::string,
-            struct base_tag,
-            st::equality_comparable,
-            st::addable_with<char *>,
-            st::addable_with<const char *>>;
+    using st_spread = st::type<
+            double,
+            struct spread_tag,
+            st::arithmetic
+            >;
 
     using st_price = st::type<
             double,
             struct price_tag,
-            st::arithmetic, st::addable_with<double>>;
+            st::arithmetic,
+            st::addable_with<double>,
+            st::multiplicable_with<double>
+            >;
+
+    using st_quantity = st::type<
+            double,
+            struct quantity_tag,
+            st::arithmetic,
+            st::addable_with<double>
+            >;
+
+    struct asset
+    {
+      st_symbol symbol;
+
+      bool operator==(const asset &rhs) const;
+      bool operator!=(const asset &rhs) const;
+    };
 
     struct pair
     {
-        st_quote quote;
-        st_base base;
+        asset quote;
+        asset base;
+
+        std::pair<asset, asset> to_std_pair();
+        bool operator==(const pair &rhs) const;
     };
+
+    enum class side { buy, sell, both };
```

Comment:
enum not class is suffisent here: 

enum class are great when you specify the underlying type like:

```cpp
enum class the_enum : uint_8 {...};
```